hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -530,7 +535,16 @@ private Object doLoad(LoadEvent event, EntityPersister persister, EntityKey keyT
 					= CacheEntityLoaderHelper.INSTANCE.loadFromSessionCache( event, keyToLoad, options );
 			final Object entity = persistenceContextEntry.getEntity();
 			if ( entity != null ) {
-				return persistenceContextEntry.isManaged() ? entity : null;
```

Comment:
Should we also backport this to 5? As far as I understand, the expectation of this method is that the result is an initialized entity, right? Not sure if not initializing it in 5 might cause the same issues as in 6, but just wanted to throw this thought in for consideration.

Reply:
I think in 5 it is not necessary